File: app/src/main/java/com/metahumanz/pacilread/model/ImportedBook.java
package com.metahumanz.pacilread.model;

import java.util.ArrayList;
import java.util.List;

public class ImportedBook {
    public String title;
    public String author;
    public String sourceDisplayName;
    public String storedPath;
    public String bookType = "text";
    public final List<ChapterSeed> chapters = new ArrayList<>();

    public static class ChapterSeed {
        public final String title;
        public final String bodyHtml;
        public final String bodyText;
        public final int orderIndex;

        public ChapterSeed(String title, String bodyHtml, String bodyText, int orderIndex) {
            this.title = title;
            this.bodyHtml = bodyHtml;
            this.bodyText = bodyText;
            this.orderIndex = orderIndex;
        }
    }
}

File: app/src/main/kotlin/com/metahumanz/pacilread/model/ImportedBook.kt
package com.metahumanz.pacilread.model

class ImportedBook {
    @JvmField var title: String? = null
    @JvmField var author: String? = null
    @JvmField var sourceDisplayName: String? = null
    @JvmField var contentSha256: String? = null
    @JvmField var storedPath: String? = null
    @JvmField var coverPath: String? = null
    @JvmField var bookType: String = "text"
    @JvmField val chapters: MutableList<ChapterSeed> = ArrayList()

    class ChapterSeed(
        @JvmField val title: String?,
        @JvmField val bodyHtml: String?,
        @JvmField val bodyText: String?,
        @JvmField val orderIndex: Int,
    )
}
